### PR TITLE
Document the `variant_pricing` tag

### PR DESCRIPTION
**Pull request by noon-ng**
As the tin says, add long-needed documentation for the `variant_pricing` tag.

### CODE REVIEW

File: docs/reference/tags/variant_pricing/index.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+title: Variant Pricing
+parent: Tags
+grand_parent: Reference
+---
+
+# Variant Pricing
+
+The `variant_pricing` tag renders a self-updating `<span>` which can
+interactively fetch and display up-to-date pricing information for a given
+variant, provided an adult count/start date/end date combination.
```
Comment (samantha-lind) [inline docs/reference/tags/variant_pricing/index.md:12]: Are these variables all required, or are some optional? I believe it also accepts other variables, can these please be listed (along with whether they are required or not)?

File: docs/reference/tags/variant_pricing/index.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+##### input
+{% raw %}
+
+    {% variant_pricing variant: variant, class: "my-nice-class", adult_count: 2, start_on: "2023-01-01", end_on: "2023-01-04" %}
+
+{% endraw %}
+
+##### output
+
+The `<span>` tag will initially be rendered empty:
```
Comment (jessehigson) [inline docs/reference/tags/variant_pricing/index.md:31]: I'm not sure this is correct? On inital page load, this should be returning a result based on the values initially passed in.

File: docs/reference/tags/variant_pricing/index.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+- Then assign the chosen values to their data-attributes 
+
+Given the `variant_pricing` tag above, one may use the following event handler:
+
+{% raw %}
+
+    function updatePrices(startOnDate, endOnDate, adultCount) {
+      const variantPricingTag = document.querySelector(".my-nice-class")
+
+      variantPricingTag.dataset.variantPricingStartOnValue = startOnDate
```
Comment (jessehigson) [inline docs/reference/tags/variant_pricing/index.md:78]: Out of context, these variables don't mean anything. Would it be better to pass in strings with the date in the accepted format to make it clearer?

File: docs/reference/tags/variant_pricing/index.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+We recommend listening to these changes on the page, and have the event handlers:
+
+- Target the `<span>` tags by their class 
+- Then assign the chosen values to their data-attributes 
+
+Given the `variant_pricing` tag above, one may use the following event handler:
+
+{% raw %}
+
+    function updatePrices(startOnDate, endOnDate, adultCount) {
```
Comment (jessehigson) [inline docs/reference/tags/variant_pricing/index.md:75]: Out of context, these variables don't mean anything. Would it be better to pass in strings with the date in the accepted format to make it clearer?

File: docs/reference/tags/variant_pricing/index.md (hunk 2 above)
Comment (noon-ng) [reply docs/reference/tags/variant_pricing/index.md:31]: Yup I mistook the use of `DOMContentLoaded` to bootstrap the content for the tag's actual behavior. I see that we've already talked about tweaking the tag so it actually does this -- let me check what comes out of standup and I'll feed back on this.

File: docs/reference/tags/variant_pricing/index.md (hunk 1 above)
Comment (noon-ng) [reply docs/reference/tags/variant_pricing/index.md:12]: Good point, will indicate more clearly which variables are required and not.

The current package step _is_ a parameter, but you can't specify it explicitly, we infer it from the page we're rendering it on.

File: docs/reference/tags/variant_pricing/index.md (hunk 2 above)
Comment (noon-ng) [reply docs/reference/tags/variant_pricing/index.md:31]: Okay, I'll update this to reflect the current behavior then revisit the docs once we fix the tag.

File: docs/reference/tags/variant_pricing/index.md (hunk 1 above)
Comment (noon-ng) [reply docs/reference/tags/variant_pricing/index.md:12]: Addressed 6861ff5

File: docs/reference/tags/variant_pricing/index.md (hunk 2 above)
Comment (noon-ng) [reply docs/reference/tags/variant_pricing/index.md:31]: Addressed 6861ff5

File: docs/reference/tags/variant_pricing/index.md (hunk 4 above)
Comment (noon-ng) [reply docs/reference/tags/variant_pricing/index.md]: Addressed 6861ff5

**Review by jessehigson — APPROVED** [review]

**Pull request merged**